### PR TITLE
Fix description for array-like fields

### LINKED ISSUE #178: Select description gets attached to multiple field
In the demo [here](https://github.com/pydantic/FastUI/blob/bb4fb360b7c2ccbae290bcd6a89d33813f6f59bb/demo/forms.py#L124) if we add a description to the first select as:
```python
class SelectForm(BaseModel):
    select_single: ToolEnum = Field(title='Select Single', description="First selector")
    select_multiple: list[ToolEnum] = Field(title='Select Multiple')
    search_select_single: str = Field(json_schema_extra={'search_url': '/api/forms/search'})
    search_select_multiple: list[str] = Field(json_schema_extra={'search_url': '/api/forms/search'})
```
it produces the following, where we can see the description leaking to the next select:
image

However, I noticed that if I use a different `Enum` for the second selector this does not happen; e.g:
```python
class ToolEnum2(str, enum.Enum):
    hammer = 'hammer'
    screwdriver = 'screwdriver'
    saw = 'saw'
    claw_hammer = 'claw_hammer'

class SelectForm(BaseModel):
    select_single: ToolEnum = Field(title='Select Single', description="First selector")
    select_multiple: list[ToolEnum2] = Field(title='Select Multiple')
    search_select_single: str = Field(json_schema_extra={'search_url': '/api/forms/search'})
    search_select_multiple: list[str] = Field(json_schema_extra={'search_url': '/api/forms/search'})
```
Screenshot 2024-02-09 at 17 59 48

Note: adding a description to the second select in the initial case doesn't work either, the first description still keeps it spot:
```python
class SelectForm(BaseModel):
    select_single: ToolEnum = Field(title='Select Single', description="First selector")
    select_multiple: list[ToolEnum] = Field(title='Select Multiple', description="Second selector")
    search_select_single: str = Field(json_schema_extra={'search_url': '/api/forms/search'})
    search_select_multiple: list[str] = Field(json_schema_extra={'search_url': '/api/forms/search'})
```
image

my 2 cents; I believe the description gets attached to the components by it's name and the component name is derived by the `Enum` name - though i am unsure how to solve it for now

### PR DESCRIPTION
**Pull request by sergue1**
Fixes #178

### CODE REVIEW

**Comment by tim-habitat** [comment]
I think this also applies to other field - in particular had an error with `json_schema` now [here](https://github.com/pydantic/FastUI/pull/177#discussion_r1486533695)

File: src/python-fastui/tests/test_forms.py [hunk 1]
```diff
@@ -469,3 +469,29 @@ def test_form_textarea_form_fields():
             }
         ],
     }
+
+
+class FormSelectMultiple(BaseModel):
+    values: List[Literal['foo', 'bar']] = Field(title='Select Multiple', description='First Selector')
```
Comment (samuelcolvin) [inline src/python-fastui/tests/test_forms.py:475]: surely we need more than one select field in the test to demonstrate that the description doesn't get leaked to subsequent inputs?

**Review by samuelcolvin — COMMENTED** [review]
otherwise LGTM.

File: src/python-fastui/tests/test_forms.py (hunk 1 above)
Comment (sergue1) [reply src/python-fastui/tests/test_forms.py]: yes and improved test revealed another source of the problem.. should have been fixed now

**Review by sydney-runkle — APPROVED** [review]
@sergue1,

Looks good, though the line on 313 isn't covered by the newly added test - is it necessary?

**Comment by sergue1** [comment]
@sydney-runkle , I am a bit confused that 313 line is uncovered: revert it and the newly added test will fail as expected. May be it is an issue with codecov?

**Comment by sydney-runkle** [comment]
Huh, odd! Yeah, maybe an issue. I'll see if I can try to skip that check for this PR.

**Comment by sydney-runkle** [comment]
Ah ok I think perhaps this is because we changed a line that was not covered before?

**Review by sydney-runkle — APPROVED** [review]
Great work!

**Pull request merged**